### PR TITLE
Allow ad-hoc code signing for TestWebKitAPI and WebKitTestRunner

### PR DESCRIPTION
**Pull request by emw-apple**
#### e347e3e762ae6078618689646e69c9d4aca2e152
<pre>
Allow ad-hoc code signing for TestWebKitAPI and WebKitTestRunner
<a href="https://bugs.webkit.org/show_bug.cgi?id=242730">https://bugs.webkit.org/show_bug.cgi?id=242730</a>

Reviewed by Alexey Proskuryakov.

Xcode 14 requires a signing identity when building command-line tools.
Internal builds already use a custom signing identity
(WK_ENGINEERING_CODE_SIGN_IDENTITY). Change the default identity for
open-source builds to ad-hoc (&quot;-&quot;).

* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
* Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig:

Prior to this change, EWS was not signing these binaries with
entitlements, even though we declare them. It turns out that many of the
entitlements we declare require a valid signing identity. While the
long-term solution might be to give all EWS machines a proper signing
identity (and ask contributors to do the same), we can remove these
entitlements from open-source builds for now. Tests that depend on the
entitlements&apos; capabilities will fail, but they were already failing.

* Tools/TestWebKitAPI/Configurations/TestWebKitAPI-macOS.entitlements:
  TestWebKitAPI already has a &quot;TestWebKitAPI-macOS-internal.entitlements&quot;
  file. Remove internal-only entitlements from this file.
* Tools/WebKitTestRunner/Configurations/WebKitTestRunner-internal.entitlements: Copied from Tools/WebKitTestRunner/Configurations/WebKitTestRunner.entitlements.
* Tools/WebKitTestRunner/Configurations/WebKitTestRunner.entitlements:
  Remove internal-only entitlements.

Canonical link: <a href="https://commits.webkit.org/254192@main">https://commits.webkit.org/254192@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/8502c389ce7af9510622b33bf1afc80cb26fe7c2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88375 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/32813 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19210 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/97559 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/153032 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/92340 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/31312 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/26962 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/80571 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92217 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/93982 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/24907 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75222 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/24881 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/79807 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/79927 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/67845 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/28923 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/13908 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/67/builds/28905 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/14931 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2949 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/32105 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/37846 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31058 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/34044 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | |

### CODE REVIEW

**Comment by aj062** [comment]
Seems like it is causing problem on api-mac, e.g.: https://ews-build.webkit.org/#/builders/3/builds/78642, https://ews-build.webkit.org/#/builders/3/builds/78640, https://ews-build.webkit.org/#/builders/3/builds/78639 etc.

I cancelled https://ews-build.webkit.org/#/builders/3/builds/78655
please check

File: Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig [hunk 1]
```diff
@@ -122,6 +122,7 @@ CODE_SIGNING_ALLOWED_YES = YES;
 CODE_SIGN_IDENTITY[sdk=macosx*] = $(CODE_SIGN_IDENTITY_$(CONFIGURATION));
 CODE_SIGN_IDENTITY_Debug = $(CODE_SIGN_IDENTITY_$(USE_INTERNAL_SDK));
 CODE_SIGN_IDENTITY_Release = $(CODE_SIGN_IDENTITY_$(USE_INTERNAL_SDK));
+CODE_SIGN_IDENTITY_ = -;
```
Comment (aproskuryakov) [inline Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig]: Is the issue here that we are building these tools as neither Debug nor Release? I wonder if that's what needs to be fixed.

**Review by aproskuryakov — CHANGES_REQUESTED** [review]
Marking as needing changes, because EWS seems unhappy.

File: Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig (hunk 1 above)
Comment (emw-apple) [reply Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig]: I don't think so. Left a comment on the bug with my understanding of what's going on (tl;dr EWS has never used the entitlements we claim it does, and turning on ad-hoc signing reveals that those entitlements require a proper signing identity): https://bugs.webkit.org/show_bug.cgi?id=242730#c11

Comment (aproskuryakov) [reply Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig]: I haven't read the comment yet. However, looking just at this one line in the PR, we wouldn't need it unless CONFIGURATION was empty in practice, correct?

Comment (emw-apple) [reply Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig]: Oh I see what you mean. This is used when USE_INTERNAL_SDK is empty.

File: Tools/TestWebKitAPI/Configurations/TestWebKitAPI-macOS.entitlements [hunk 2]
```diff
@@ -2,15 +2,6 @@
...
 <dict>
-	<key>keychain-access-groups</key>
-	<array>
-		<string>com.apple.TestWebKitAPIAlternate</string>
-		<string>com.apple.TestWebKitAPI</string>
-	</array>
-	<key>com.apple.private.webkit.webpush</key>
-	<true/>
-	<key>com.apple.private.webkit.webpush.inject</key>
-	<true/>
```
Comment (aproskuryakov) [inline Tools/TestWebKitAPI/Configurations/TestWebKitAPI-macOS.entitlements]: The commit message says that we can remove entitlements from open source build, but this seems to remove them altogether. I understand the WKTR change, but not the TestWebKitAPI one.

File: Tools/WebKitTestRunner/Configurations/WebKitTestRunner-internal.entitlements [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>keychain-access-groups</key>
+	<array>
+		<string>com.apple.WebKitTestRunner</string>
+	</array>
+	<key>com.apple.private.webkit.adattributiond.testing</key>
+	<true/>
+	<key>com.apple.security.temporary-exception.sbpl</key>
```
Comment (aproskuryakov) [inline Tools/WebKitTestRunner/Configurations/WebKitTestRunner-internal.entitlements]: Duplicating the entitlements is of course dangerous. If producing them with a script like we do in WebKit itself is too much hassle, perhaps the opensource version should be renamed to `WebKitTestRunner-opensource.entitlements`, as a prompt that it's not the only file to modify?

File: Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig [hunk 4]
```diff
@@ -43,11 +43,16 @@ EXCLUDED_SOURCE_FILE_NAMES_ = ;
 EXCLUDED_SOURCE_FILE_NAMES_cocoatouch = mac/*;
 INCLUDED_SOURCE_FILE_NAMES = mac/main.mm;
 
+MAC_CODE_SIGN_ENTITLEMENTS = $(MAC_CODE_SIGN_ENTITLEMENTS_$(USE_INTERNAL_SDK));
+MAC_CODE_SIGN_ENTITLEMENTS_ = Configurations/WebKitTestRunner.entitlements;
+MAC_CODE_SIGN_ENTITLEMENTS_YES = Configurations/WebKitTestRunner-internal.entitlements;
+
 CODE_SIGN_ENTITLEMENTS[sdk=macosx*] = Configurations/WebKitTestRunner.entitlements;
```
Comment (aproskuryakov) [inline Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig]: `MAC_CODE_SIGN_ENTITLEMENTS` seems to remain unused.

**Review by aproskuryakov — APPROVED** [review]

File: Tools/WebKitTestRunner/Configurations/WebKitTestRunner-internal.entitlements (hunk 3 above)
Comment (emw-apple) [reply Tools/WebKitTestRunner/Configurations/WebKitTestRunner-internal.entitlements]: I don't think it's high-risk for slipping out of sync, as any Apple engineer will need to update the internal entitlements for their at-desk builds, and the open source entitlements for any added tests to pass EWS.

re: naming, the `-internal` suffix is precedented by `Tools/TestWebKitAPI/Configurations/TestWebKitAPI-macOS-internal.entitlements`. I'd prefer matching that convention, but if you really want to change it we should change it in both places.

File: Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig (hunk 4 above)
Comment (emw-apple) [reply Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig]: Good catch, thank you!

File: Tools/TestWebKitAPI/Configurations/TestWebKitAPI-macOS.entitlements (hunk 2 above)
Comment (emw-apple) [reply Tools/TestWebKitAPI/Configurations/TestWebKitAPI-macOS.entitlements]: There is a separate, pre-existing `Tools/TestWebKitAPI/Configurations/TestWebKitAPI-macOS-internal.entitlements` which provides internal-only entitlements. The changes to WKTR make it behave more like TestWebKitAPI in this area.

File: Tools/WebKitTestRunner/Configurations/WebKitTestRunner-internal.entitlements (hunk 3 above)
Comment (aproskuryakov) [reply Tools/WebKitTestRunner/Configurations/WebKitTestRunner-internal.entitlements]: I think that it's better for both to have suffixes (`-internal` and `-opensource`), as having no suffix for one of them is confusing. It confused me when discussing this PR.

Would be even better to generate them.

This is not a blocker for landing this PR.

**Review by aproskuryakov — APPROVED** [review]
r=ews

**Pull request merged**